prometheus.lua

```diff
@@ -355,6 +403,31 @@ function Prometheus:inc(name, label_names, label_values, value)
   self:log_error_kv(key, value, err)
 end
 
+-- Set the current value of a gauge to `value`
+-- Args:
+--   name: (string) short metric name without any labels.
+--   label_names: (array) a list of label keys.
+--   label_values: (array) a list of label values.
+--   value: (number) the new value for the gauge.
+function Prometheus:set_gauge(name, label_names, label_values, value)
```

Comment:
This is not really gauge-specific (i.e. you could use this method to set a value to a counter as well, even if you probably should not). Could you call it `Prometheus:set()` and rename existing `set` method to `set_key`, please?